python/pyiceberg/expressions/__init__.py

```diff
@@ -281,6 +289,30 @@ def __invert__(self) -> BoundIsNull:
         return BoundIsNull(self.term)
 
 
+def coerce_unary_arguments(data_class: Union[type, None]) -> Union[type, Callable[[type], type]]:
```

Comment:
@Fokko, this PR introduces 3 of these decorators to make it easy to construct predicates. This decorator replaces `__init__` with a version that converts a string passed as `term` into a `Reference`. This makes it much easier to create expressions, as you can see from the test cases.

The other decorators handle set predicates and literal predicates.

Reply:
I'm not sold on this, mostly because it adds another layer of complexity. These things should normally just happen in the init. I've created a PR https://github.com/apache/iceberg/pull/6127 That does exactly this, well tested 👍🏻